modules/data.atmosphere/R/download.NEONmet.R

```diff
@@ -0,0 +1,486 @@
+      } else { nairTemp3 = 0 }
+    if ((length(airTempLoc)==0) && (length(airTemp3Loc)==0)) {
+      logger.error("Air temperature DP1.00002 or DP1.00003 not available") }
+    if ((nairTemp==0) && (nairTemp3==0)) {
+      logger.error("Air temperature DP1.00002 or DP1.00003 not available in date range ",startMon," ",endMon)
+    }
+  
+  #define NetCDF variable and create NetCDF file
+    airT.var <- ncvar_def(name = "air_temperature", units = "K", dim = xytdim)
+    nc <- ncdf4::nc_create(new.file, vars = airT.var)  #create netCDF file
+    ncdata <- rep(FillValue,length(datetime))
+    if (nairTemp3>nairTemp) {
+      for (mon in airTemp3Dates[airTemp3GoodDates]) {
+        neonData <- nneo::nneo_data(product_code = availProducts[airTemp3Loc[1]], site_code = site, year_month = mon)
+        if (length(neonData$data$urls)>0) {
+          #Extract and read 30 minute data from the highest vertical level among files returned
+          url30 <- tail(sort(neonData$data$urls[grep("*.00000.000.*_30min",neonData$data$urls)]),1)
+          csvData <- read.csv(url30,stringsAsFactors=FALSE,header=TRUE)
+          #Retreive time dimension and figure out where in array to put it
+          csvDateTime <- as.POSIXct(gsub("T"," ",csvData$startDateTime),tz="UTC")
+          arrLoc <- floor(as.numeric(difftime(csvDateTime,datetime[1],tz="UTC",units="hours"))*2)+1
+          csvVar <- csvData$tempTripleMean
+          csvQF <- csvData$finalQF
+          csvVar[which(csvQF!=1)]=NA  #triple-asp temp always has QF of 1
+          ncdata[arrLoc] <- udunits2::ud.convert(csvVar,"celsius", "K")
+        }
+      }
+    } else {
+      for (mon in airTempDates[airTempGoodDates]) {
+        neonData <- nneo::nneo_data(product_code = availProducts[airTempLoc[1]], site_code = site, year_month = mon)
+        if (length(neonData$data$urls)>0) {
+          #Extract and read 30 minute data from the highest vertical level among files returned
+          url30 <- tail(sort(neonData$data$urls[grep("*.00000.000.*_30min",neonData$data$urls)]),1)
+          csvData <- read.csv(url30,stringsAsFactors=FALSE,header=TRUE)
+          #Retreive time dimension and figure out where in array to put it
+          csvDateTime <- as.POSIXct(gsub("T"," ",csvData$startDateTime),tz="UTC")
+          arrLoc <- floor(as.numeric(difftime(csvDateTime,datetime[1],tz="UTC",units="hours"))*2)+1
+          csvVar <- csvData$tempSingleMean
+          csvQF <- csvData$finalQF
+          csvVar[which(csvQF!=0)]=NA
+          ncdata[arrLoc] <- udunits2::ud.convert(csvVar,"celsius", "K")
+        }
+      }
+    }
+    ncdf4::ncvar_put(nc, varid = airT.var, vals = ncdata)
+
+    # NEON.DP1.00004 Pressure
+    pSurfLoc = grep("DP1.00004",availProducts)
+    if (length(pSurfLoc)!=0) { 
+      pSurfDates = unlist(availDates[pSurfLoc[1]])
+      pSurfGoodDates = which((pSurfDates >= startMon) & (pSurfDates <= endMon))
+      npSurf = length(pSurfGoodDates)
+    } else {npSurf=0}
```

Comment:
Assignment operators here. Also, not the dot in the `grep` regex, as above. Also, `else` block should be on its own line.